### ISSUE TITLE
Site Editor: Clarify the mysterious 'Entity' 👻  messages.

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#47886]
### Description

In the Site Editor, a couple messages used for the Snackbar and for the accessible audible messages refer to an 'Entity' 👻 .
It is not known whether that is a philosophical, metaphysical entity, a supernatural entity, or any other ontological-related concept.

- `Entity renamed.`
- `Entity reverted.`

Jokes apart, user-facing messages should avoid developers jargon. Such messages should be simple, concise, self-explanatory.

In other places in the editor, similar messages do distinguish the 'entity' in question and craft a simpler. more meaningful message based on whether the edited object is a `template` or `template part`. Similarly, these strings should be refactored to provide cleared messages for users.

Aside:
from a visual perspective, the Snackbar with dark gray background shows up on top of the sidebar with dark gray background so that the Snackbar shape is barely distinguishable. Not sure that's great. This should go in a separate issue though.

### Step-by-step reproduction instructions

- Go to the Site editor.
- Create a custom template.
- Click Templates > Manage all templates in the navigation.
- In the templates list, click the Actions button of the custom template.
- Choose Rename, change the template name and save.
- Observe a Snackbar appears at the bottom left of the window.
- Observe the Snackbar text is `Entity renamed.`.
- The same text is used for an accessible `speak` message for screen reader users.
- Make a change to an existing default template.
- Click Templates > Manage all templates in the navigation.
- In the templates list, click the Actions button of the customized template.
- Click Clear customizations.
- Observe a Snackbar appears at the bottom left of the window.
- Observe the Snackbar text is `Entity reverted.`.
- The same text is used for an accessible `speak` message for screen reader users.

### Screenshots, screen recording, code snippet

<img width="1787" alt="Entity renamed" src="https://user-images.githubusercontent.com/1682452/217581604-d7ab6897-ba47-42f2-9336-ddae1d67d925.png">

<img width="1750" alt="Entity reverted" src="https://user-images.githubusercontent.com/1682452/217581628-a44997c7-aec3-4e1a-bddb-546c78a4fdbf.png">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Divyansh1908** [comment]
Do you want it to be changed to **template renamed** and **template reset** ?

**Comment by afercia** [comment]
@Divyansh1908 thanks for looking into this.

Ideally, the messages should distinguish whether it's a template or a template part. Something along these lines:

```
const message =
	template.type === 'wp_template'
		? __( 'Template renamed.' )
		: __( 'Template part renamed.' );

createSuccessNotice( message, {
	type: 'snackbar',
} );
```

Same goes for the 'reverted' messages.

Also, I just noticed there are two more 'entity' messages:
- `An error occurred while reverting the entity.`
- `An error occurred while renaming the entity.`

These could be simplified by just removing the words ` the entity.`

**Comment by Divyansh1908** [comment]
Ok, Thank you for clarifying it for me. I am currently very new to open-source contributions and hence clarifications like these make it easier to understand. 
I will try resolving the issue with these hints

**Comment by prakhar2407** [comment]
@afercia Assign this issue to me, I will do it

**Comment by afercia** [comment]
Not sure who of you two guys I should assign this issue to 🙂 Not sure I'm supposed to 'assign' anyone in the first place, as I'm just a contributor amongst other contributors. @Divyansh1908 came first. I'd suggest to be nice each other and follow the spirit of OSS, where anyone is free to propose their own contributions. If you need any guidance, you may want to go through the [contributor guide](https://github.com/WordPress/gutenberg/tree/HEAD/CONTRIBUTING.md).

**Comment by jordesign** [comment]
I cam across this in my testing of 6.3 beta 4 - it's more relevant (I think) now as the messaging is shown for both Templates and Patterns. Having clarity of what has been renamed/reverted here would be nice.

@prakhar2407 did you end up making any progress on this?

**Comment by jordesign** [comment]
Adding a video for clarity:


https://github.com/WordPress/gutenberg/assets/1842363/b4cf2183-9ab4-4a80-94cd-f2019f3b9dd5

**Comment by annezazu** [comment]
Since this doesn't relate to 6.3 exactly and we're hitting RC1, punting this to 6.3.x. It's not necessary for this release.
